bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/FileRecoveryLog.java

```diff
@@ -0,0 +1,237 @@
+package io.ballerina.runtime.transactions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.file.Path;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class FileRecoveryLog implements RecoveryLog {
+    private static final Logger log = LoggerFactory.getLogger(FileRecoveryLog.class);
+    private String baseFileName;
+    private Path recoveryLogDir;
+    private int checkpointInterval;
+    private final boolean deleteOldLogs;
+    private int numOfPutsSinceLastCheckpoint;
+    private File file;
```

Comment:
Shall we rename this to `logFile` because we have used `file` in some function parameters in this class as well.